File: app/src/main/java/com/m6world/tari_labs/api/TimeOutParameters.java
package com.m6world.tari_labs.api;

public  class TimeOutParameters {
    public final static int CONNECT_DEFAULT = 8, WRITE_DEFAULT = 32, READ_DEFAULT = 96;
    public int connect = CONNECT_DEFAULT, write = WRITE_DEFAULT, read = READ_DEFAULT;

    public TimeOutParameters(int connect, int write, int read) {
        this(connect, write);
        if (read > 0) this.read = read;
    }

    public TimeOutParameters(int connect, int write) {
        this(connect);
        if (write > 0) this.write = write;
    }

    public TimeOutParameters(int connect) {
        if (connect > 0) this.connect = connect;
    }
}

File: app/src/main/java/com/m6world/tari_labs/api/TimeOutParameters.kt
package com.m6world.tari_labs.api

class TimeOutParameters {
    var connect = CONNECT_DEFAULT
    var write = WRITE_DEFAULT
    var read = READ_DEFAULT

    constructor(connect: Int, write: Int, read: Int) : this(connect, write) {
        if (read > 0) this.read = read
    }

    constructor(connect: Int, write: Int) : this(connect) {
        if (write > 0) this.write = write
    }

    constructor(connect: Int) {
        if (connect > 0) this.connect = connect
    }

    companion object {
        val CONNECT_DEFAULT = 8
        val WRITE_DEFAULT = 32
        val READ_DEFAULT = 96
    }
}